app/models/service_orchestration/provision_tagging.rb

```diff
@@ -22,11 +22,19 @@ def provisioning_tag_ids
     provision_sequence = miq_request_task.provision_priority + 1
     dialog_options = root_service.options[:dialog] || {}
 
+    tags = []
     dialog_options.flat_map do |key_name, value|
       if (match = DIALOG_TAG_KEY_REGEX.match(key_name))
         tag_sequence = match[:sequence].to_i
-        value.scan(DIALOG_TAG_VALUE_REGEX).flatten if tag_sequence.zero? || tag_sequence == provision_sequence
+        if value.kind_of?(Array)
+          value.each do |tag|
+            tags.push(tag.scan(DIALOG_TAG_VALUE_REGEX).flatten) if tag_sequence.zero? || tag_sequence == provision_sequence
+          end
+        else
+          tags.push(value.scan(DIALOG_TAG_VALUE_REGEX).flatten) if tag_sequence.zero? || tag_sequence == provision_sequence
+        end
```

Comment:
Minor, but could you `Array.wrap(value).each ...` and only have one code path here and reduce the duplication?

Reply:
Should be fixed now